arch/cortex-m/src/dwt.rs

```diff
@@ -0,0 +1,525 @@
+
+        /// Reads as one if a second linked comparator is supported.
+        LNK1ENA     OFFSET(9)    NUMBITS(1),
+
+        /// Enables data value comparison
+        /// When 0: Perform address comparison, when 1: data value comparison.
+        /// RW.
+        DATAVMATCH  OFFSET(8)    NUMBITS(1),
+
+        /// Write 1 to enable generation of data trace address packets.
+        /// WARN: If `Control::NOTRCPKT` reads as zero, this bit is UNKNOWN.
+        /// RW.
+        EMITRANGE   OFFSET(5)    NUMBITS(1),
+
+        /// Selects action taken on comparator match.
+        /// Resets to 0b0000.
+        /// RW.
+        FUNCTION    OFFSET(0)    NUMBITS(4),
+    ],
+
+    Comparator3[
+        /// Reference value for comparator 0.
+        COMP       OFFSET(0)   NUMBITS(32),
+    ],
+
+    Comparator3Mask[
+        /// Size of ignore mask aplied to the access address for address range matching by comparator 0.
+        /// WARN: Maximum Mask size is IMPLEMENTATION DEFINED.
+        MASK       OFFSET(0)   NUMBITS(5),
+    ],
+
+    Comparator3Function[
+        /// Is one if comparator matches. Reading the register clears it to 0.
+        /// RO.
+        MATCHED     OFFSET(24)   NUMBITS(1),
+
+        /// Second comparator number for linked address comparison.
+        /// Works, when `DATAVMATCH` and `LNK1ENA` read as one.
+        /// RW.
+        DATAVADDR1  OFFSET(16)   NUMBITS(4),
+
+        /// Comparator number for linked address comparison.
+        /// Works, when `DATAVMATCH` reads as one.
+        /// RW.
+        DATAVADDR0  OFFSET(12)   NUMBITS(4),
+
+        /// Size of data comparision (Byte, Halfword, Word).
+        /// RW.
+        DATAVSIZE   OFFSET(10)   NUMBITS(2),
+
+        /// Reads as one if a second linked comparator is supported.
+        LNK1ENA     OFFSET(9)    NUMBITS(1),
+
+        /// Enables data value comparison
+        /// When 0: Perform address comparison, when 1: data value comparison.
+        /// RW.
+        DATAVMATCH  OFFSET(8)    NUMBITS(1),
+
+        /// Write 1 to enable generation of data trace address packets.
+        /// WARN: If `Control::NOTRCPKT` reads as zero, this bit is UNKNOWN.
+        /// RW.
+        EMITRANGE   OFFSET(5)    NUMBITS(1),
+
+        /// Selects action taken on comparator match.
+        /// Resets to 0b0000.
+        /// RW.
+        FUNCTION    OFFSET(0)    NUMBITS(4),
+    ],
+
+];
+
+const DWT: StaticRef<DwtRegisters> = unsafe { StaticRef::new(0xE0001000 as *const DwtRegisters) };
+
+struct Dwt {
+    registers: StaticRef<DwtRegisters>,
+}
+/// Returns wether a cycle counter is present on the chip.
+pub fn is_cycle_counter_present() -> bool {
+    if DWT.ctrl.read(Control::NOCYCCNT) == 0 {
+        true
+    } else {
+        false
+    }
+}
+
+impl Dwt {
+    pub const fn new() -> Self {
+        Self { registers: DWT }
+    }
+}
+
+impl hil::debug::PerformanceCounters for Dwt {
+    fn enable_cycle_counter() -> Result<(), ErrorCode> {
```

Comment:
Personally I would call this `start_cycle_counter()` since it actually starts it counting up, rather than just enabling the functionality. Same for `disable_cycle_counter()` -- I would just call that `stop_cycle_counter()`, and not have it disable the DCB.